code/modules/research/designs.dm

```diff
@@ -2439,6 +2439,15 @@ k
 	materials = list("$iron" = 500, "$glass" = 1000, "$gold" = 200, "$silver" = 200)
 	build_path = /obj/item/device/device_analyser
 
+/datum/design/component_exchanger
+	name = "Rapid Machinery Component Exchanger"
+	desc = "A device that allows to quickly replace machinery components, useful for upgrading."
+	id = "deviceanalyser"

```

Comment:
No joke, seriously everyone fucking does this same mistake, it makes the object not load into the design lists properly (as in, at all)